### PR TITLE
Remove caching of global window side input param when invoking.

### PR DESCRIPTION
**Pull request by scwhittle**
Instead rely on the higher-level cache which utilizes the side input cache tokens allowing the runner to determine the caching used for side inputs.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by scwhittle** [comment]
Addresses #28776

**Review by riteshghorse — APPROVED** [review]
LGTM, Thanks!

CC: @AnandInguva

File: sdks/python/apache_beam/runners/common.py [hunk 1]
```diff
@@ -1086,6 +1053,9 @@ def _invoke_process_batch_per_window(
 
     kwargs_for_process_batch = kwargs_for_process_batch or {}
 
+    if additional_kwargs:
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/common.py:1056]: Is this a drive-by change or is it directly related to the intention of this PR?

File: sdks/python/apache_beam/runners/common.py [hunk 2]
```diff
@@ -926,31 +922,16 @@ def _invoke_process_per_window(self,
...
-      side_inputs.extend(additional_args)
-      args_for_process, kwargs_for_process = util.insert_values_in_args(
-          self.args_for_process, self.kwargs_for_process,
-          side_inputs)
-    elif self.cache_globally_windowed_args:
-      # Attempt to cache additional args if all inputs are globally
-      # windowed inputs when processing the first element.
-      self.cache_globally_windowed_args = False
-
-      # Fill in sideInputs if they are globally windowed
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/common.py:943]: Is my understanding correct that the fundamental flaw in this logic is that even for a global window, the sideinput can refresh if a window has a trigger configured?
